semver/compare.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { SemVer } from "./semver.ts";
+import { parse } from "./parse.ts";
+
+/**
+ * Compare two semantic version objects.
+ *
+ * Returns `0` if `v1 == v2`, or `1` if `v1` is greater, or `-1` if `v2` is
+ * greater.
+ *
+ * Sorts in ascending order if passed to `Array.sort()`,
+ */
+export function compare(
+  s0: SemVer,
+  s1: SemVer,
+): 1 | 0 | -1;
+/** @deprecated (will be removed after 0.189.0) Use `compare(s0: SemVer, s1: SemVer)` instead. */
+export function compare(
+  s0: string | SemVer,
+  s1: string | SemVer,
+  options?: { includePrerelease: boolean },
+): 1 | 0 | -1;
+export function compare(
+  s0: string | SemVer,
+  s1: string | SemVer,
+  options?: { includePrerelease: boolean },
+): 1 | 0 | -1 {
+  const v0 = typeof s0 === "string" ? parse(s0, options) : s0;
+  const v1 = typeof s1 === "string" ? parse(s1, options) : s1;
```

Comment:
It seems like parse itself supports strings so why not just call it and have the logic to differentiate between string and SemVer in that one spot?

```suggestion
  const v0 = parse(s0, options);
  const v1 = parse(s1, options);
```

Reply:
Thanks. This makes sense.